File: coyote/apis/msgs_stream.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    MsgStreamCommitIn,
    MsgStreamCommitOut,
    MsgStreamReceiveIn,
    MsgStreamReceiveOut,
    MsgStreamSeekIn,
    MsgStreamSeekOut,
)

from ..models.msg_stream_receive_in import _MsgStreamReceiveIn
from ..models.msg_stream_commit_in import _MsgStreamCommitIn
from ..models.msg_stream_seek_in import _MsgStreamSeekIn


class MsgsStreamAsync(ApiBase):
    async def receive(
        self,
        topic: str,
        consumer_group: str,
        msg_stream_receive_in: MsgStreamReceiveIn = MsgStreamReceiveIn(),
    ) -> MsgStreamReceiveOut:
        """Receives messages from a topic using a consumer group.

        Each consumer in the group reads from all partitions. Messages are locked by leases for the
        specified duration to prevent duplicate delivery within the same consumer group."""
        body = _MsgStreamReceiveIn(
            namespace=msg_stream_receive_in.namespace,
            topic=topic,
            consumer_group=consumer_group,
            batch_size=msg_stream_receive_in.batch_size,
            lease_duration_ms=msg_stream_receive_in.lease_duration_ms,
            default_starting_position=msg_stream_receive_in.default_starting_position,
            batch_wait_ms=msg_stream_receive_in.batch_wait_ms,
        ).model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.msgs.stream.receive",
            body=body,
            response_type=MsgStreamReceiveOut,
        )

    async def commit(
        self,
        topic: str,
        consumer_group: str,
        msg_stream_commit_in: MsgStreamCommitIn,
    ) -> MsgStreamCommitOut:
        """Commits an offset for a consumer group on a specific partition.

        The topic must be a partition-level topic (e.g. `ns:my-topic~3`). The offset is the last
        successfully processed offset; future receives will start after it."""
        body = _MsgStreamCommitIn(
            namespace=msg_stream_commit_in.namespace,
            topic=topic,
            consumer_group=consumer_group,
            offset=msg_stream_commit_in.offset,
        ).model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.msgs.stream.commit",
            body=body,
            response_type=MsgStreamCommitOut,
        )

    async def seek(
        self,
        topic: str,
        consumer_group: str,
        msg_stream_seek_in: MsgStreamSeekIn = MsgStreamSeekIn(),
    ) -> MsgStreamSeekOut:
        """Repositions a consumer group's read cursor on a topic.

        Provide exactly one of `offset` or `position`. When using `offset`, the topic must include a
        partition suffix (e.g. `ns:my-topic~0`). The `position` field accepts `"earliest"` or
        `"latest"` and may be used with or without a partition suffix."""
        body = _MsgStreamSeekIn(
            namespace=msg_stream_seek_in.namespace,
            topic=topic,
            consumer_group=consumer_group,
            offset=msg_stream_seek_in.offset,
            position=msg_stream_seek_in.position,
        ).model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.msgs.stream.seek",
            body=body,
            response_type=MsgStreamSeekOut,
        )


class MsgsStream(ApiBase):
    def receive(
        self,
        topic: str,
        consumer_group: str,
        msg_stream_receive_in: MsgStreamReceiveIn = MsgStreamReceiveIn(),
    ) -> MsgStreamReceiveOut:
        """Receives messages from a topic using a consumer group.

        Each consumer in the group reads from all partitions. Messages are locked by leases for the
        specified duration to prevent duplicate delivery within the same consumer group."""
        body = _MsgStreamReceiveIn(
            namespace=msg_stream_receive_in.namespace,
            topic=topic,
            consumer_group=consumer_group,
            batch_size=msg_stream_receive_in.batch_size,
            lease_duration_ms=msg_stream_receive_in.lease_duration_ms,
            default_starting_position=msg_stream_receive_in.default_starting_position,
            batch_wait_ms=msg_stream_receive_in.batch_wait_ms,
        ).model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.msgs.stream.receive",
            body=body,
            response_type=MsgStreamReceiveOut,
        )

    def commit(
        self,
        topic: str,
        consumer_group: str,
        msg_stream_commit_in: MsgStreamCommitIn,
    ) -> MsgStreamCommitOut:
        """Commits an offset for a consumer group on a specific partition.

        The topic must be a partition-level topic (e.g. `ns:my-topic~3`). The offset is the last
        successfully processed offset; future receives will start after it."""
        body = _MsgStreamCommitIn(
            namespace=msg_stream_commit_in.namespace,
            topic=topic,
            consumer_group=consumer_group,
            offset=msg_stream_commit_in.offset,
        ).model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.msgs.stream.commit",
            body=body,
            response_type=MsgStreamCommitOut,
        )

    def seek(
        self,
        topic: str,
        consumer_group: str,
        msg_stream_seek_in: MsgStreamSeekIn = MsgStreamSeekIn(),
    ) -> MsgStreamSeekOut:
        """Repositions a consumer group's read cursor on a topic.

        Provide exactly one of `offset` or `position`. When using `offset`, the topic must include a
        partition suffix (e.g. `ns:my-topic~0`). The `position` field accepts `"earliest"` or
        `"latest"` and may be used with or without a partition suffix."""
        body = _MsgStreamSeekIn(
            namespace=msg_stream_seek_in.namespace,
            topic=topic,
            consumer_group=consumer_group,
            offset=msg_stream_seek_in.offset,
            position=msg_stream_seek_in.position,
        ).model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.msgs.stream.seek",
            body=body,
            response_type=MsgStreamSeekOut,
        )

File: src/api/msgs_stream.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct MsgsStream<'a> {
    cfg: &'a Configuration,
}

impl<'a> MsgsStream<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Receives messages from a topic using a consumer group.
    ///
    /// Each consumer in the group reads from all partitions. Messages are locked by leases for the
    /// specified duration to prevent duplicate delivery within the same consumer group.
    pub async fn receive(
        &self,
        topic: String,
        consumer_group: String,
        msg_stream_receive_in: MsgStreamReceiveIn,
    ) -> Result<MsgStreamReceiveOut> {
        let msg_stream_receive_in = MsgStreamReceiveIn_ {
            namespace: msg_stream_receive_in.namespace,
            topic,
            consumer_group,
            batch_size: msg_stream_receive_in.batch_size,
            lease_duration_ms: msg_stream_receive_in.lease_duration_ms,
            default_starting_position: msg_stream_receive_in.default_starting_position,
        };

        crate::request::Request::new(http::Method::POST, "/api/v1.msgs.stream.receive")
            .with_body(msg_stream_receive_in)
            .execute(self.cfg)
            .await
    }

    /// Commits an offset for a consumer group on a specific partition.
    ///
    /// The topic must be a partition-level topic (e.g. `ns:my-topic~3`). The offset is the last
    /// successfully processed offset; future receives will start after it.
    pub async fn commit(
        &self,
        topic: String,
        consumer_group: String,
        msg_stream_commit_in: MsgStreamCommitIn,
    ) -> Result<MsgStreamCommitOut> {
        let msg_stream_commit_in = MsgStreamCommitIn_ {
            namespace: msg_stream_commit_in.namespace,
            topic,
            consumer_group,
            offset: msg_stream_commit_in.offset,
        };

        crate::request::Request::new(http::Method::POST, "/api/v1.msgs.stream.commit")
            .with_body(msg_stream_commit_in)
            .execute(self.cfg)
            .await
    }

    /// Repositions a consumer group's read cursor on a topic.
    ///
    /// Provide exactly one of `offset` or `position`. When using `offset`, the topic must include a
    /// partition suffix (e.g. `ns:my-topic~0`). The `position` field accepts `"earliest"` or
    /// `"latest"` and may be used with or without a partition suffix.
    pub async fn seek(
        &self,
        topic: String,
        consumer_group: String,
        msg_stream_seek_in: MsgStreamSeekIn,
    ) -> Result<MsgStreamSeekOut> {
        let msg_stream_seek_in = MsgStreamSeekIn_ {
            namespace: msg_stream_seek_in.namespace,
            topic,
            consumer_group,
            offset: msg_stream_seek_in.offset,
            position: msg_stream_seek_in.position,
        };

        crate::request::Request::new(http::Method::POST, "/api/v1.msgs.stream.seek")
            .with_body(msg_stream_seek_in)
            .execute(self.cfg)
            .await
    }
}
